libs/wlgen/wlgen/rta.py

```diff
@@ -186,72 +186,28 @@ def __postrun(self, params):
 
 
     def _getFirstBig(self, cpus=None):
-        cpus = cpus or range(self.target.number_of_cpus)
-
-        if 'bl'in self.target.modules:
-            cluster = self.target.bl.bigs
-            candidates = sorted(set(self.target.bl.bigs).intersection(cpus))
-            if candidates:
-                return cadidates[0]
 
         return cpus[0]
 
-    def _getFirstLittle(self, cpus=None):
-        # Non big.LITTLE system:
-        if 'bl' not in self.target.modules:
-            # return the first CPU of the first cluster
-            platform = self.target.platform
-            cluster_first = list(set(platform.core_clusters))[0]
-            cluster_cpus = [cpu_id
-                    for cpu_id, cluster_id in enumerate(platform.core_clusters)
-                                           if cluster_id == cluster_first]
-            # If CPUs have been specified': return the fist in the first cluster
-            if cpus:
-                for cpu_id in cpus:
-                    if cpu_id in cluster_cpus:
-                        return cpu_id
-            # Otherwise just return the first cpu of the first cluster
-            return cluster_cpus[0]
-
-        # Try to return one LITTLE CPUs among the specified ones
-        if cpus:
-            for c in cpus:
-                if c not in self.target.bl.littles:
-                    continue
-                return c
-        # Only big CPUs, thus:
-        #  return the first LITTLE core of the system
-        if self.target.little_core:
-            # Big.LITTLE system
-            return self.target.bl.littles[0]
-        return 0
-
-    def getTargetCpu(self, loadref):
+    def getCalibrationConf(self):
         # Select CPU for task calibration, which is the first little
         # of big depending on the loadref tag
         if self.pload is not None:
-            if loadref and loadref.upper() == 'LITTLE':
-                target_cpu = self._getFirstLittle()
-                self._log.debug('ref on LITTLE cpu: %d', target_cpu)
+            if self.loadref and self.loadref.upper() == 'LITTLE':
+                return max(self.pload.values())
             else:
-                target_cpu = self._getFirstBig()
-                self._log.debug('ref on big cpu: %d', target_cpu)
-            return target_cpu
-
-        # These options are selected only when RTApp has not been
-        # already calibrated
-        if self.cpus is None:
-            target_cpu = self._getFirstBig()
-            self._log.debug('ref on cpu: %d', target_cpu)
+                return min(self.pload.values())
```

Comment:
It returns the value to set for the 'calibration' key in the rt-app JSON. [This can be either a string like "CPU0" or an integer representing the pLoad value](https://github.com/scheduler-tools/rt-app/blob/master/doc/tutorial.txt#L90).